### PR TITLE
Update architecture and sequence diagrams with waveform information

### PR DESCRIPTION
**Pull request by markmatney**

### CODE REVIEW

**Review by DRickard — DISMISSED** [review]

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -40,12 +40,10 @@ Actions (e.g., the parsing of CSV files, conversion of media files, or storage o
 
 ## Sequence Diagram
 
-The sequence diagram below illustrates the order of events that happen as the av-pairtree program processes an A/V file and puts it into a [Pairtree](https://tools.ietf.org/html/draft-kunze-pairtree-01) directory structure.
+The sequence diagram below illustrates the order of events that happen as the av-pairtree program processes a CSV:
```
Comment (ksclarke) [inline ARCHITECTURE.md:43]: You don't think it's useful to link to the Pairtree spec?

Comment (markmatney) [reply ARCHITECTURE.md:43]: Oh, for some reason I didn't realize I'd removed the link. I do think it's useful but maybe earlier in the README?

File: docs/images/architecture.mermaid [hunk 2]
```diff
@@ -1,26 +1,75 @@
...
+                    activate PairtreeVerticle
+
+                    PairtreeVerticle->>File System: Reads temporary MP4 audio file
+                    PairtreeVerticle-->>ConverterVerticle: Notifies of successful Pairtree insertion
+                    deactivate PairtreeVerticle
+
+                    ConverterVerticle->>File System: Deletes temporary MP4 audio file
+                    ConverterVerticle-->>WatcherVerticle: Notifies of successful conversion and insertion
+                    deactivate ConverterVerticle
+                and Generates waveform data and stores on Amazon S3
```
Comment (ksclarke) [inline docs/images/architecture.mermaid:44]: Is this supposed to be left indented farther than the others?

File: ARCHITECTURE.md (hunk 1 above)
Comment (markmatney) [reply ARCHITECTURE.md:43]: er, [it's already linked in the "Bird's Eye View" section](https://github.com/UCLALibrary/av-pairtree/pull/10/files#diff-8f6366fd8e7a5fa30f7f05879999195b7cf5fa1e3d157822eb37f0e91d226cfdR9) -- that's enough IMO

**Comment by ksclarke** [comment]
Is there any way to include a dictionary to explain what "alt" and "par" are? There is more going on now and I feel like it's harder to read. In that vein, any way to highlight the headers (like "[is video]") so that they're easier to spot?

File: docs/images/architecture.mermaid (hunk 2 above)
Comment (markmatney) [reply docs/images/architecture.mermaid:44]: Based on [the syntax description of the `par ... and` construct](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=parallel), it looks right to me (there isn't a concrete example like there is for `alt ... else` or `loop`, but the examples for those constructs use the same indentation pattern that I've used here).

File: ARCHITECTURE.md (hunk 1 above)
Comment (ksclarke) [reply ARCHITECTURE.md:43]: Okay 👍🏼

File: docs/images/architecture.mermaid (hunk 2 above)
Comment (ksclarke) [reply docs/images/architecture.mermaid:44]: Ah, okay... part of the problem, I think, is that alt and par aren't obvious (I don't think one should have to go read about the diagramming spec to know what the components of the diagram are).

**Comment by markmatney** [comment]
I agree that the increased complexity makes it harder to read. What if under the diagram, we link to the [alt](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=alt) and [par](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=parallel) sections of the docs? I can also experiment with the styling/configuration of the diagram.

**Comment by ksclarke** [comment]
> I agree that the increased complexity makes it harder to read. What if under the diagram, we link to the [alt](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=alt) and [par](https://mermaid-js.github.io/mermaid/#/sequenceDiagram?id=parallel) sections of the docs? I can also experiment with the styling/configuration of the diagram.

I think that's better, but still not ideal.

**Comment by markmatney** [comment]
One other route would be to extract those nested sequences into separate diagrams.

**Comment by ksclarke** [comment]
> One other route would be to extract those nested sequences into separate diagrams.

That might be better / more readable

**Comment by markmatney** [comment]
How's this? I tried to apply CSS to the diagram, but [mermaid doesn't actually support that](https://github.com/mermaid-js/mermaid/search?q=styling&type=issues).

**Comment by ksclarke** [comment]
I like the audio sequence. Not crazy about the big yellow box on the main sequence. I know doing one for video and one for audio would be repeating some of the sequence, but what do you think about that approach?

I'm also not sure if I'm being too picky about this (and whether it's worth a ton of work to make it look nice) though. So, if you want to push back, feel free.

**Comment by markmatney** [comment]
Just pushed another commit, which I think looks a bit better. Unfortunately with the auto-generated nature of these diagrams (and the buggy-ness of mermaid) I think our formatting options are few.

**Review by ksclarke — APPROVED** [review]
I agree the split version is better / simpler to read.

**Review by DRickard — COMMENTED** [review]
FWIW, GitHub is complaining about lack of newline endings in sequence_audio.svg and sequence_video.mermaid.

**Comment by markmatney** [comment]
The SVGs are auto-generated (and not human readable anyway) so I think it's fine

**Review by DRickard — APPROVED** [review]

**Pull request merged**